game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/firststrike/DefensiveFirstStrike.java

```diff
@@ -99,15 +110,27 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
     if (this.state == State.NOT_APPLICABLE) {
       return;
     }
-    battleActions.findTargetGroupsAndFire(
-        returnFire,
-        battleState.getPlayer(OFFENSE).getName() + SELECT_FIRST_STRIKE_CASUALTIES,
-        true,
-        battleState.getPlayer(DEFENSE),
-        Matches.unitIsFirstStrikeOnDefense(battleState.getGameData()),
-        battleState.filterUnits(ALIVE, DEFENSE),
-        battleState.filterUnits(CASUALTY, DEFENSE),
-        battleState.filterUnits(ALIVE, OFFENSE),
-        battleState.filterUnits(CASUALTY, OFFENSE));
+    final List<BattleStep> steps = getSteps();
+
+    // steps go in reverse order on the stack
```

Comment:
This seems familiar ;)
Perhaps the `FireStepsBuilder` should know this and just return a stack itself?